disnake/state.py

```diff
@@ -1911,18 +1911,66 @@ def _get_reaction_user(
             return channel.guild.get_member(user_id)
         return self.get_user(user_id)
 
-    def get_reaction_emoji(self, data) -> Union[Emoji, PartialEmoji]:
-        emoji_id = utils._get_as_snowflake(data, "id")
+    # methods to handle all sorts of different emoji formats
+
+    def _get_emoji_from_data(
+        self, data: PartialEmojiPayload
+    ) -> Optional[Union[str, Emoji, PartialEmoji]]:
```

Comment:
how can this return None? I don't see a codepath that allows that.

Reply:
it's not immediately apparent, but `data["name"]` is nullable